content/ui-patterns/saving.mdx

```diff
@@ -0,0 +1,222 @@
+
+Explicitly saved forms can be saved by clicking a submit button or by pressing Enter while focused on an input.
+
+If there is an error saving the data, the user's data should be preserved in the form and they should be given [feedback](#feedback) about the failure.
+
+<DoDontContainer>
+  <Do>
+    <img alt="A form broken into two sections where one section has a dropdown that automatically saves" src="https://user-images.githubusercontent.com/2313998/151887935-980adacd-41aa-47cb-8756-eeaa1a6d0f2b.png" />
+    <Caption>Separate auto-saving controls from an explicitly saved form</Caption>
+  </Do>
+  <Dont>
+    <img alt="A form with a single section and it contains a dropdown that automatically saves" src="https://user-images.githubusercontent.com/2313998/151887936-d41900ca-2d63-43ac-bf82-55b4bba0df61.png" />
+    <Caption>Mix auto-saving controls with explictly saved controls in the same form</Caption>
+  </Dont>
+</DoDontContainer>
+
+### Unsaved changes
+
+With the addition of the command palette there are additional opportunities to accidentally navigate away from a page with unpublished edits.
+
+If a user tries to navigate away from a page with a form that has unsaved changes, you can optionally warn them that their changes will be lost using the [`beforeunload` event](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event). Modern browsers prevent developers from customizing the `alert` message, so we're limited to a generic message like "Do you want to leave this site? Changes you made may not be saved."
+
+### Declarative controls
+
+- Text inputs
+- Checkbox groups
+- Radio button groups
+- Browser-native `<select>` dropdown menus
+- Multi-select dropdown menus
+
+To maintain consistent interaction patterns across your app, do not use segmented controls in an explicitly saved form. These should only be used with automatic saving.
+
+To ensure your forms are accessible, do not automatically save declarative inputs.
+
+#### Problems with autosaving declarative inputs:
+**Text inputs**
+- Users are expecting to set a value and then submit
+- Sensitive information could be unintentionally submitted. For example: a change password input
+
+**Checkbox groups**
+- It could be unclear whether or not their selections have been saved
+- Users may accidentally submit a selection that they clicked by accident
+
+**Radio button groups**
+- Screenreader users can't read the options without selecting them. This could accidentally apply a selection the user didn't want.
+
+**Browser-native `<select>` dropdowns**
+- A select dropdown has a similar problem as a radio button group. Options are selected when a user focuses the select and uses their arrow keys to read through each option
+
+<DoDontContainer>
+  <Do>
+    <img src="https://user-images.githubusercontent.com/2313998/151623711-ad294946-fd67-428d-96b2-e4f534857c8a.png" />
```

Comment:
`@vdepizzol` has updated menu mocks here: https://www.figma.com/file/n56A2QJsjdFnkwuBvX9Jdz/save-button-in-select-panel?node-id=0%3A14382